Kim Edits-HL7-MLLP (#1324)
mulesoft/docs-connectors: hl7-mllp/2.1/modules/ROOT/edits/hl7-mllp-connector-studio.adoc

> = Use Anypoint Studio to Configure HL7 MLLP Connector - Mule 4

Reviewer: This topic only skimmed, assuming adherence to the boilerplate text.

. On the *Listener* configuration screen, optionally change the value of the *Display Name* field.
. Specify a value for the *Path* field.
. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of the HTTP Listener source in the app.
. On the *General* tab, specify connection information for the connector.
. On the *TLS* tab, optionally specify the TLS information for the connector.
. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
. Click *Test Connection* to confirm that Mule can connect with the specified server.
. Click *OK*.

[[add-connector-operation]]
== Add a Connector Operation to the Flow

When you add a connector operation to your flow, you are specifying an action for that connector to perform.

To add an operation for HL7 EDI Connector, follow these steps:

. In the *Mule Palette* view, select *HL7 MLLP* and then select the desired operation.
. Drag the operation onto the Studio canvas to the right of the input source.

[[configure-global-element]]
== Configure a Global Element for the Connector

> When you configure a connector, configure a global element that all instances of that connector in the app can use. Configuring a global element requires you to provide the authentication credentials that the connector requires to access the target HL7 MLLP system.

Reviewer: Note that "global element" is lowercased.

Author: STET in template.

To configure the global element for HL7 MLLP Connector, follow these steps:

. Select the operation in the Studio canvas.
. In the configuration screen for the operation, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.
+
You can reference a configuration file that contains ANT-style property placeholders (recommended), or you can enter your authorization credentials in the global configuration properties.

. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
. Click *Test Connection* to confirm that Mule can connect with the specified server.
. Click *OK*.

> You also can create a TLS configuration, you can set the timeout configuration and the buffer configuration.

Reviewer: This is not a complete sentence: refer to boilerplate or otherwise fix this.

Author: STET in template.

> The Send operation the procedure is identical, except that you can set the Local Address Settings tab within the connector config with a different host, port, and delimiter that specifies the character used to separate messages.

Reviewer: Here is my best guess at what this sentence means to say; please reread and fix:
```suggestion
The Send operation of the procedure is identical, except that you can configure the *Local Address Settings* tab to use a different host, port, and delimiter that specifies the character used to separate messages.
```

Author: STET in template.

You can also set the send or receive buffer size.

> NOTE: If your app is running at mllp-rnnf.us-e1.cloudhub.io, you can send HL7 messages over MLLP to your Mule app to mule-worker-mllp-rnnf.us-e1.cloudhub.io

Reviewer: ```suggestion
NOTE: If your app is running at `mllp-rnnf.us-e1.cloudhub.io`, you can send HL7 messages over MLLP to your Mule app to `mule-worker-mllp-rnnf.us-e1.cloudhub.io`.
```

Author: fixed.

== Next Step

> You can try the xref:hl7-mllp-connector-examples.adoc[Examples].

Reviewer: Note wording: try the example rather than use the example.

Author: STET in template.

== See Also

* xref:hl7-mllp-connector-reference.adoc[HL7 MLLP Connector Reference]

> * https://help.mulesoft.com[MuleSoft Help Center]

Reviewer: BREAK to commit

Reviewer: RESUME to commit #2
